### PR TITLE
Remove the draft prefix to ERC20Permit

### LINKED ISSUE #3790: Remove the draft prefix for `ERC20Permit` and `IERC20Permit` as it is now final
**🧐 Motivation**

I just realised that `EIP-2612` is finally "Final" ([PR](https://github.com/ethereum/EIPs/pull/5782)). Thx for pushing that through @frangio! I quickly opened this issue to track the renaming (in a backward-compatible way of course like for `EIP-712`; https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3621) of [`draft-ERC20Permit.sol`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/draft-ERC20Permit.sol) and [`draft-IERC20Permit.sol`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/draft-IERC20Permit.sol) to `ERC20Permit.sol` and `IERC20Permit.sol`.

### PR DESCRIPTION
**Pull request by Amxx**
Fixes #3790

#### PR Checklist

- [ ] Tests
- [ ] Documentation
- [ ] Changelog entry

### CODE REVIEW

**Comment by pcaversaccio** [comment]
I actually started working on a PR without realising that @Amxx has already done it :-D. In any case, I wanted to make you aware of that in `docs/modules/ROOT/pages/governance.adoc` & `scripts/migrate-imports.js` we probably also need to make the adjustments. See my commit here:
- https://github.com/pcaversaccio/openzeppelin-contracts/commit/0326538a392bd2b4cceb7416347424604f78525b#diff-fe8d0766559dcc7458e0b283082155dc4647977692e8385d1b04931fcdbbe402
- https://github.com/pcaversaccio/openzeppelin-contracts/commit/0326538a392bd2b4cceb7416347424604f78525b#diff-be3eea41b81c2bebec062279c3b97653515ff424d4595df2f8a3598192423877

I'm not yet sure about the usage of `scripts/migrate-imports.js` but we probably would need as well need the following lines:

```js
'cryptography/EIP712.sol': 'utils/cryptography/EIP712.sol', // Your forgot to update iirc
'token/ERC20/ERC20Permit.sol': 'token/ERC20/extensions/ERC20Permit.sol',
'token/ERC20/IERC20Permit.sol': 'token/ERC20/extensions/IERC20Permit.sol',
```

**Comment by Amxx** [comment]
We are keeping the old files (as redirect to the new) for backward compatibility ... so forgetting that would not have been a major issue ... but you are right, fixing that is better !

File: contracts/interfaces/draft-IERC2612.sol [hunk 1]
```diff
@@ -1,8 +1,8 @@
 // SPDX-License-Identifier: MIT
-// OpenZeppelin Contracts v4.4.1 (interfaces/draft-IERC2612.sol)
 
 pragma solidity ^0.8.0;
 
-import "../token/ERC20/extensions/draft-IERC20Permit.sol";
+// IERC2612 was marked as draft due to the EIP-2612 dependency.
```
Comment (frangio) [inline contracts/interfaces/draft-IERC2612.sol:5]: This message was added in ERC721Votes to explain that it was a draft due to EIP712 (712, not 721!), so it's not relevant in this case.

```suggestion
```

File: contracts/token/ERC20/extensions/draft-ERC20Permit.sol [hunk 2]
```diff
@@ -1,95 +1,8 @@
...
-// OpenZeppelin Contracts (last updated v4.6.0) (token/ERC20/extensions/draft-ERC20Permit.sol)
 
 pragma solidity ^0.8.0;
 
-import "./draft-IERC20Permit.sol";
-import "../ERC20.sol";
-import "../../../utils/cryptography/ECDSA.sol";
-import "../../../utils/cryptography/EIP712.sol";
-import "../../../utils/Counters.sol";
+// IERC20Permit was marked as draft due to the EIP-2612 dependency.
```
Comment (frangio) [inline contracts/token/ERC20/extensions/draft-ERC20Permit.sol:5]:
```suggestion
```

File: contracts/token/ERC20/extensions/draft-IERC20Permit.sol [hunk 3]
```diff
@@ -1,60 +1,8 @@
...
-    function permit(
-        address owner,
-        address spender,
-        uint256 value,
-        uint256 deadline,
-        uint8 v,
-        bytes32 r,
-        bytes32 s
-    ) external;
+// IERC20Permit was marked as draft due to the EIP-2612 dependency.
```
Comment (frangio) [inline contracts/token/ERC20/extensions/draft-IERC20Permit.sol:5]:
```suggestion
```

**Review by frangio — APPROVED** [review]

**Comment by cpecorari** [comment]
Hi, just seen that 4.8.0 version does not include final ERC20Permit, is that a mistake because draft-ERC20Permit.sol file states :
```
// SPDX-License-Identifier: MIT
// OpenZeppelin Contracts (last updated v4.8.0) (token/ERC20/extensions/draft-ERC20Permit.sol)

pragma solidity ^0.8.0;

// EIP-2612 is Final as of 2022-11-01. This file is deprecated.

import "./ERC20Permit.sol";
```

Thanks !

**Comment by pcaversaccio** [comment]
@cpecorari no that's correct - you have to check the released version here: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.8.0/contracts/token/ERC20/extensions/draft-ERC20Permit.sol. I guess the release `4.9.0` will include this change - otherwise use OpenZeppelin via submodules to use the latest `master` version.

**Comment by frangio** [comment]
Indeed this didn't make it to 4.8. But please do not use the `master` branch, this is highly discouraged because the release process contributes to the security of the code.

There is nothing different in the code, so you can use the "draft" version from 4.8.

**Pull request merged**